### PR TITLE
feat(client): add jsdocs for apis

### PR DESCRIPTION
**Pull request by Mister-Hope**

### CODE REVIEW

**Comment by Mister-Hope** [comment]
I think we'd better make some improvments before merging this pr. including:

- [ ] A lot of api is marked as void results, but I think the server may return errorcode and we are not handling these kindof situations.
- [ ] `getArticleCounter` really have an awful return type, and I think it will confuse anyone who want to make a custom client with our apis. I think we should provide new apis with different url that have a better result and remain the old one. So we may get rid of this in V3 client.

**Comment by Mister-Hope** [comment]
CC @lizheming

**Comment by Mister-Hope** [comment]
Also I would like to share some preference:

```ts
value as type
```

is preferred if the value is actually having a type cast.

And 

```ts
<Type>value
```

is preferred if the type anonation is "marking" the value from "any" or "unknown" returned by any api or something which is not providing a clear type as expected.

**Comment by lizheming** [comment]
> I think we'd better make some improvments before merging this pr. including:
> 
> * [ ]  A lot of api is marked as void results, but I think the server may return errorcode and we are not handling these kindof situations.
> * [ ]  `getArticleCounter` really have an aweful return type, and I think it will confuse anyone who want to make a custom client with our apis. I think we should provide new apis with different url that have a better result and remain the old one. So we may get rid of this in V3 client.

Most of API have a wrapper like `{"err_no": 0, "err_msg": "", "data": ...}` except:

1. comment List for client: `/comment`
2. recent comment list: `/comment?type=recent`
3. comment count: `/comment?type=count`
4. get and post article: `/article`

I agree with you we should unify return structures. But it's a break change for old version user, do you have any good idea? I think we can add a `version` params for API.

**Review by lizheming — APPROVED** [review]

**Comment by Mister-Hope** [comment]
My idea is to add new apis for counter with a constant return inteface. Also, we can improve api by adding new path for backword compatability.

E.g.:

- A new `/commentData` besides `/comment`
- A new `/counter` besides `/article`

I am not sure all the env will return  `{"errno":404,"errmsg":"url `/xxx` not found."}`  or not with unexisit path, if so, I can make compatable version for old server in V2.

Client can completely drop old apis in v3 while vercel can drop those in v2.

**Comment by Mister-Hope** [comment]
Any feedback about proposal api? @lizheming

**Comment by lizheming** [comment]
> Any feedback about proposal api? @lizheming

I'll modify it in the next big version, those discussion will be reference.

File: packages/client/src/api/comment.ts [hunk 1]
```diff
@@ -38,53 +89,62 @@ export const fetchComment = ({
...
   };
 
   if (token) headers.Authorization = `Bearer ${token}`;
 
-  if (comment.eid) {
-    return fetch(`${serverURL}/comment/${comment.eid}?lang=${lang}`, {
+  return fetch(
+    `${serverURL}/comment${comment.eid ? `/${comment.eid}` : ''}?lang=${lang}`,
+    {
       method: 'PUT',
```
Comment (lizheming) [inline packages/client/src/api/comment.ts:140]: @Mister-Hope 为什么这块变成了 PUT 请求了呀？

**Pull request merged**